docs/DevGuide/DevGuide.md

```diff
@@ -4,21 +4,42 @@ See LICENSE.txt for details.
 \endcond
 # Developer's Guide {#dev_guide}
 
+### Developing and Improving Executables
 - \ref spectre_build_system "Build system" and how to add dependencies,
   unit tests, and executables.
-- \ref code_review_guide "Code review guidelines." All code merged into
-  master must follow these requirements.
-- \ref code_concepts "Concepts" used throughout the code are defined here
-  for reference.
 - \ref dev_guide_creating_executables "Executables and how to add them"
 - \ref dev_guide_option_parsing "Option parsing" to get options from input files
-- \ref ParallelGroup "Parallelization infrastructure" components and usage
 - \ref profiling_with_projections "Profiling With Charm++ Projections" and PAPI
   for optimizing performance
-- \ref template_metaprogramming "Template Metaprogramming"
-- \ref travis_guide "Travis CI"
-- \ref writing_good_dox "Writing good documentation" is key for long term
-  maintainability of the project
 - \ref spectre_writing_python_bindings "Writing Python Bindings" to use
   SpECTRE C++ classes and functions from within python.
-- \ref writing_unit_tests "Writing Unit Tests"
+
+### Having your Contributions Merged into SpECTRE
+- \ref writing_good_dox "Writing good documentation" is key for long term
+  maintainability of the project.
+- \ref writing_unit_tests "Writing Unit Tests" to catch bugs and to make
+  sure future changes don't cause your code to regress.
+- \ref travis_guide "Travis CI" is used to test every pull request.
+- \ref code_review_guide "Code review guidelines." All code merged into
+  develop must follow these requirements.
+
+### General SpECTRE Concepts
```

Comment:
_General concepts_ and _foundational concepts_ seems not too different. Maybe call this __SpECTRE Terminology__